apps/mobile/src/state/accounts/accounts.slice.ts

```diff
@@ -1,26 +1,52 @@
 import { useSelector } from 'react-redux';
 
+import { AvatarIcon } from '@/components/avatar-icon';
+import { t } from '@lingui/macro';
 import { createAction, createEntityAdapter, createSelector, createSlice } from '@reduxjs/toolkit';
+import { produce } from 'immer';
 
 import type { RootState } from '..';
 import { handleAppResetWithState, userAddsWallet, userRemovesWallet } from '../global-action';
 import { BitcoinKeychainStore } from '../keychains/bitcoin/bitcoin-keychains.slice';
 import { StacksKeychainStore } from '../keychains/stacks/stacks-keychains.slice';
-import { handleEntityActionWith, makeAccountIdentifer } from '../utils';
+import { Optional, handleEntityActionWith, makeAccountIdentifer } from '../utils';
 import { initalizeAccount } from './accounts';
 
-type AccountStatus = 'deleted';
+type AccountStatus = 'active' | 'hidden';
 
 export interface AccountStore {
   id: string;
-  name?: string;
-  status?: AccountStatus;
+  icon: string | AvatarIcon;
+  name: string;
+  status: AccountStatus;
 }
 
 const adapter = createEntityAdapter<AccountStore, string>({
   selectId: account => account.id,
 });
 
+function addAccountDefaults({
+  account,
+  accountIdx,
+}: {
+  account: PartialAccountStore;
+  accountIdx: number;
+}): AccountStore {
+  const updatedAccount = produce(account, draftAccount => {
+    if (!draftAccount.icon) {
+      draftAccount.icon = 'sparkles';
+    }
+    if (!draftAccount.name) {
+      draftAccount.name = t`Account ${accountIdx}`;
+    }
+    if (!draftAccount.status) {
+      draftAccount.status = 'active';
+    }
+    return draftAccount;
+  });
+  return updatedAccount as AccountStore;
+}
```

Comment:
Is the immer code here needed? Isn't this handled internally by RTK entity methods?

Also see the `accountIdx` is for order, but would suggest another name, orderBy or something, just because "account index" is already a used concept for it's path account index